### PR TITLE
feat: always declare a ts_project _types target to fetch dts files

### PR DESCRIPTION
**Pull request by jbedard**
Sometimes the dts files need to be fetched, there is no reason this shouldn't be done when a transpiler is not being used.

This makes the declare targets more predictable.

---

### Changes are visible to end-users: yes

- Searched for relevant documentation and updated as needed: yes
- Breaking change (forces users to change their own code or config): no
- Suggested release notes appear below: yes

`ts_project(name)` declaration files can be fetched using the `{name}_types` target.

### Test plan

- Covered by existing test cases
- New test cases added

### CODE REVIEW

File: examples/transpiler/BUILD.bazel [hunk 1]
```diff
@@ -93,6 +101,27 @@ build_test(
...
 
+# Runs babel to transpile ts -> js while tsc would fail with typecheck errors
+# Tagging as manual should prevent the typecheck from running even when requesting
+# transpiler output files.
+ts_project(
+    name = "typecheck_fail",
+    srcs = ["typecheck_fail.ts"],
+    declaration = True,
+    source_map = True,
+    tags = ["manual"],
```
Comment (jbedard) [inline examples/transpiler/BUILD.bazel:112]: This test is extra, I just want to make sure failing tsc is not invoked when tagged as manual, even when the non-tsc transpiler is forced to be invoked.

File: examples/targets/BUILD.bazel [hunk 2]
```diff
@@ -35,17 +35,36 @@ ts_project(
...
         ":gen-a-ts",
     ],
+    declaration = True,
     out_dir = "out",
+    source_map = True,
+    tsconfig = "tsconfig.json",
 )
 
-# Ensure all the outputs are pre-declared
+# Ensure all the outputs are pre-declared and in the correct locations including
```
Comment (dzbarsky) [inline examples/targets/BUILD.bazel:44]: unfinished sentence?

**Review by dzbarsky — APPROVED** [review]

**Comment by dzbarsky** [comment]
@jbedard disabled auto merge so you an address the comment

**Review by gregmagolan — APPROVED** [review]

File: examples/targets/BUILD.bazel (hunk 2 above)
Comment (jbedard) [reply examples/targets/BUILD.bazel:44]: thank you lol

**Comment by jfirebaugh** [comment]
The docs in https://github.com/aspect-build/rules_ts/blob/main/docs/transpiler.md#macro-expansion should be updated to say that the `[name]_types` expansion is always generated.

**Pull request merged**